hassio/src/dialogs/network/dialog-hassio-network.ts

```diff
@@ -0,0 +1,293 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  query,
+  internalProperty,
+  TemplateResult,
+} from "lit-element";
+import { cache } from "lit-html/directives/cache";
+
+import {
+  updateNetworkInterface,
+  NetworkInterface,
+} from "../../../../src/data/hassio/network";
+import { fireEvent } from "../../../../src/common/dom/fire_event";
+import { HassioNetworkDialogParams } from "./show-dialog-network";
+import { haStyleDialog } from "../../../../src/resources/styles";
+import { showAlertDialog } from "../../../../src/dialogs/generic/show-dialog-box";
+import type { HomeAssistant } from "../../../../src/types";
+import type { PolymerChangedEvent } from "../../../../src/polymer-types";
+
+import "../../../../src/components/ha-circular-progress";
+import "../../../../src/components/ha-dialog";
+import "../../../../src/components/ha-dialog";
+import "../../../../src/components/ha-header-bar";
+import "../../../../src/components/ha-header-bar";
+import "../../../../src/components/ha-related-items";
+import "../../../../src/components/ha-svg-icon";
+
+@customElement("dialog-hassio-network")
+export class DialogHassioNetwork extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _prosessing: boolean = false;
+
+  @internalProperty() private _params?: HassioNetworkDialogParams;
+
+  @internalProperty() private _network!: {
+    interface: string;
+    data: NetworkInterface;
+  }[];
+
+  @internalProperty() private _curTabIndex: number = 0;
+
+  @query("#ip_address") private _ip_address?: PaperInputElement;
+
+  @query("#gateway") private _gateway?: PaperInputElement;
+
+  @query("#nameservers") private _nameservers?: PaperInputElement;
+
+  public async showDialog(params: HassioNetworkDialogParams): Promise<void> {
+    this._params = params;
+    this._network = Object.keys(params.network?.interfaces)
+      .map((device) => ({
+        interface: device,
+        data: params.network.interfaces[device],
+      }))
+      .sort((a, b) => {
+        return a.data.primary > b.data.primary ? -1 : 1;
+      });
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    this._prosessing = false;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params || !this._network) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        .heading=${true}
+        hideActions
+        @closed=${this.closeDialog}
+        @close-dialog=${this.closeDialog}
+      >
+        <div slot="heading">
+          <ha-header-bar>
+            <mwc-icon-button slot="navigationIcon" dialogAction="cancel">
+              <ha-svg-icon .path=${mdiClose}></ha-svg-icon>
+            </mwc-icon-button>
+            <span slot="title">
+              Network settings
+            </span>
+          </ha-header-bar>
```

Comment:
Why don't you use `createCloseHeading`?

Reply:
I modeled it after the device dialog, but can change that ;)

Reply:
Because of the tabs!

Reply:
Right... nevermind then :smile: